core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -309,6 +328,48 @@ private void releaseBlockLockIfUnused(long blockId) {
   }
 
   /**
+   * Checks the internal state of the manager to make sure invariants hold.
+   *
+   * This method is intended for testing purposes. A runtime exception will be thrown if invalid
+   * state is encountered.
+   */
+  public void validate() {
```

Comment:
I totally agree with you on the cost to maintain reflection between production and test code. And I equally share the concern as I already see the problems caused by breaking one side without updating the other side. I guess I am thinking here that we need a coherent story on unit testing (e.g., what is the suggested way testing internal states, I am even fine with a hybrid approach of exposing "internal" method `public` and also use `PrivateAccess`, as long as we have a clear description when to use which). But before we have a story, it might be good to keep a style consistent with the current practice? And in the case we decided to stay away from reflection approach, we could convert the code together with less confusion.